### ISSUE TITLE
mobilenet v2 pretrained model?

### ISSUE THREAD

**Issue by yyjabidintg** [KaiyangZhou/deep-person-reid#116]
do you have mobilenet v2 pretrained model?
Thanks a lot

**Comment by KaiyangZhou** [comment]
do you mean imagenet pretrained?

**Comment by KaiyangZhou** [comment]
I have updated the mobilenetv2 code to make it clearer (not pushed the commit yet).

I am training mobilenetv2-1.0 and mobilenetv2-1.4 on imagenet. Will provide the imagenet weights and reid weights soon.

**Comment by KaiyangZhou** [comment]
**Update**
- Mobilenetv2 is pretrained on imagenet with batch size of 2048 and 8 GPUs. The results are

| Model | Top1 (paper) | Top1 ([tensorflow](https://github.com/tensorflow/models/tree/master/research/slim/nets/mobilenet#imagenet--checkpoints)) | Top1 (ours) |
| :---: | :---: | :---: | :---: |
| mobilenetv2-1.0 | 72.0 | 71.8 | 71.3 |
| mobilenetv2-1.4 | 74.7 | 75.0 | 73.9 |

- The reid weights have been released, please refer to the model zoo. Only the `.pth.tar` files are attached. If you wanna reproduce the results, try
```
python train_imgreid_xent.py \
-s market1501 \
-t market1501 \
-j 8 \
--height 256 \
--width 128 \
--optim amsgrad \
--label-smooth \
--lr 0.0006 \
--max-epoch 80 \
--stepsize 40 60 \
--fixbase-epoch 10 \
--open-layers classifier \
--train-batch-size 64 \
--test-batch-size 100 \
--arch mobilenetv2_1dot0 \
--save-dir log/mobilenetv2_1dot0-market \
--gpu-devices 0
```

The imagenet pretrained weights will be automatically downloaded and loaded.

**Comment by yyjabidintg** [comment]
thanks for your quick response.

**Comment by d-li14** [comment]
@yyjabidintg you can refer to https://github.com/d-li14/mobilenetv2.pytorch for a much stronger mobilenetv2 backbone

**Comment by amandayeyan** [comment]
Hi, not sure whether training epoch 80 is not enough? After fine-tune with above script, what I get is Rank-1@72.7%, and mAP:51.8%.
Any comments?

**Comment by KaiyangZhou** [comment]
@amandayeyan what command did you use?

**Comment by amandayeyan** [comment]
@KaiyangZhou 
The args I followed are exactly the same as you recommended for fine-tune. 
![image](https://user-images.githubusercontent.com/18046926/54876609-c1a86e00-4e4d-11e9-93f5-576a2f9617d1.png)

**Comment by KaiyangZhou** [comment]
that's weird, your result seems to be too low

is the pretrained weight loaded successfully?

**Comment by amandayeyan** [comment]
After 10 epoch of fixed-base training, the accuracy remains at 3.12, far below the expected 80+ or 90+. 
I am trying to figure out whether the pretrained weights loaded successfully or not. 
 
![image](https://user-images.githubusercontent.com/18046926/54891005-df75e180-4ee6-11e9-850e-511780411c80.png)

**Comment by amandayeyan** [comment]
@KaiyangZhou 
The pretrained weights were not loaded successfully because name of keys do not matach. 
In pretrained weights, keys are named as "module.xxxx" but name of keys in model_dict does not start with "module.". So I renamed the keys in pretrained weights by adding the following codes to ./torchreid/models/mobilenetv2.py.  
And remember to _import collections_

Now the training results are Rank-1@86.4 and mAP@67.2 for testing on market1501 dataset. 

![image](https://user-images.githubusercontent.com/18046926/54893903-089c6f00-4ef3-11e9-94b6-218b16d95373.png)

**Comment by KaiyangZhou** [comment]
I see. I forgot to change the key names in mobilenetv2's weight file.
